### PR TITLE
Base templating implementation; scion

### PR DESCRIPTION
**Pull request by tjerman**

### CODE REVIEW

File: store/rdbms/rdbms_schema.go [hunk 1]
```diff
@@ -238,6 +239,23 @@ func (Schema) Labels() *Table {
...
+func (Schema) Templates() *Table {
+	return TableDef("templates",
+		ID,
+		ColumnDef("rel_owner", ColumnTypeIdentifier),
+		ColumnDef("handle", ColumnTypeVarchar, ColumnTypeLength(handleLength)),
+		ColumnDef("language", ColumnTypeText),
+		ColumnDef("type", ColumnTypeText),
+		ColumnDef("partial", ColumnTypeBoolean),
+		ColumnDef("meta", ColumnTypeJson),
+		ColumnDef("template", ColumnTypeText),
```
Comment (darh) [inline store/rdbms/rdbms_schema.go:251]: Are we safe to assume that template will always be text?
Would we be better of with wrapping it into struct/doc and save it as json?

File: store/rdbms/templates.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	"github.com/cortezaproject/corteza-server/pkg/filter"
+	"github.com/cortezaproject/corteza-server/system/types"
+)
+
+func (s Store) convertTemplateFilter(f types.TemplateFilter) (query squirrel.SelectBuilder, err error) {
+	query = s.templatesSelectBuilder()
+
+	query = filter.StateCondition(query, "tpl.deleted_at", f.Deleted)
+
+	// @todo labels?
```
Comment (darh) [inline store/rdbms/templates.go]: do it. takes a couple of copy-pastes and it works.

File: store/rdbms/templates.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+	if f.TemplateID > 0 {
```
Comment (darh) [inline store/rdbms/templates.go]: Might be smart if just switch to slice of uint64 for (primary) ids.

File: store/templates.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+      It also returns deleted templates.
+  - fields: [ Handle ]
+    filter: { DeletedAt: nil }
+    uniqueConstraintCheck: true
+    description: |-
+        searches for template by the handle
+
+        It returns only valid templates (not deleted)
+  - fields: [ Language ]
```
Comment (darh) [inline store/templates.yaml]: this returns only one resource (hence "lookup"). Is this really what you need?

**Review by darh — CHANGES_REQUESTED** [review]

File: store/rdbms/rdbms_schema.go (hunk 1 above)
Comment (tjerman) [reply store/rdbms/rdbms_schema.go:251]: I... don't know...

We will be implementing some sort of a version control system for our templates so this part will need to be reworked. I didn't find much sense in complicating this part (yet) as only HTML and plaintext templates will be supported for now (plain text stuff).

I would leave it as is for now, and we can overengineer the structure a bit later when we need to.

File: store/templates.yaml (hunk 4 above)
Comment (tjerman) [reply store/templates.yaml]: Whoops

**Pull request merged**